parsec/core/gui/register_user_dialog.py

```diff
@@ -128,28 +134,29 @@ def closeEvent(self, event):
             event.accept()
 
     def register_user(self):
-        def _run_registration(username, token, is_admin):
-            self.portal.run(
-                _handle_invite_and_create_user,
-                self.register_queue,
-                self.on_registered,
-                self.on_register_error,
-                self.core.device,
-                username,
-                token,
-                is_admin,
-            )
-
         if not self.line_edit_username.text():
             show_warning(self, _("Please enter a username."))
             return
 
-        users = self.portal.run(self.core.fs.backend_cmds.user_find, self.line_edit_username.text())
+        try:
+            users = self.portal.run(
+                self.core.fs.backend_cmds.user_find, self.line_edit_username.text()
+            )
+        except BackendNotAvailable as exc:
+            show_error(
+                self,
+                QCoreApplication.translate(
+                    "RegisterUserDialog", f"Can not invite this user ({str(exc)})."
```

Comment:
"Cannot" plutôt que "Can not" ?

Et le message devrait plutôt expliquer que l'opération n'est pas possible en hors ligne

Et finalement le plus important: il faut déplacer cette opération dans le job `_do_registration`, l'idée étant d'avoir un seul job qui prend les valeurs de la GUI tel quelle, puis réalise toutes les opération métier (conversions, appel aux apis, retour d'un résultat utilisable par la GUI etc.) et enfin envoi un signal d'erreur ou de succès gérer par la GUI pour afficher le résultat

Reply:
Si je veux y appeler mon `user_find` à l'intérieur, il faut que je passe cette méthode en paramètre manifestement, car ne peux appeler direct via un import.
La GUI obtient le `user_find` via un `self.core.fs.backend_cmds.user_find`. Mais si je tente ce passage en paramètre je me prend un `connection refused` lors de l'appel au backend.

Reply:
> il faut que je passe cette méthode en paramètre manifestement

passe plutôt l'objet `self.core` (histoire que ce soit plus générique)

> Mais si je tente ce passage en paramètre je me prend un connection refused lors de l'appel au backend.

Étonnant, tu as un stacktrace ?

Reply:
En fait c'est bon ça fonctionne le souci ne venait pas de ça.